src/main/scala/firrtl/Compiler.scala

```diff
@@ -105,20 +105,28 @@ object AnnotationSeq {
   def apply(xs: Seq[Annotation]) = new AnnotationSeq(xs.toList)
 }
 
+/** Metadata associated with a Firrtl circuit
+  *
+  * @param annotations The current collection of [[firrtl.annotations.Annotation Annotation]]
+  * @param renames A map of [[firrtl.annotations.Named Named]] things that have been renamed.
+  *        Generally only a return value from [[Transform]]s
+  */
+case class Metadata(annotations: Seq[Annotation], renames: Option[RenameMap])
+object Metadata {
+  def apply(annotations: Seq[Annotation]) = new Metadata(annotations, None)
+}
+
 /** Current State of the Circuit
   *
   * @constructor Creates a CircuitState object
   * @param circuit The current state of the Firrtl AST
   * @param form The current form of the circuit
-  * @param annotations The current collection of [[firrtl.annotations.Annotation Annotation]]
-  * @param renames A map of [[firrtl.annotations.Named Named]] things that have been renamed.
-  *   Generally only a return value from [[Transform]]s
+  * @param metadata Holds auxiliary information related to the circuit
   */
 case class CircuitState(
     circuit: Circuit,
     form: CircuitForm,
-    annotations: AnnotationSeq,
-    renames: Option[RenameMap]) {
+    metadata: Metadata) {
```

Comment:
What do you think about pulling form into Metadata? I think it makes sense, as its actually metadata on the circuit.